### DISCUSSION TITLE
Error in infer_shapes.py

### DISCUSSION THREAD

**Discussion by MFJI** [Xilinx/finn#679] [Q&A]
Hello everyone,

I have noticed that I get the following error when running the infer_shapes transform: 
`  File "/workspace/qonnx/src/qonnx/transformation/infer_shapes.py", line 88, in apply
    model = ModelWrapper(si.infer_shapes(model.model))
  File "/opt/conda/lib/python3.8/site-packages/onnx/shape_inference.py", line 41, in infer_shapes
    inferred_model_str = C.infer_shapes(model_str, check_type, strict_mode, data_prop)
onnx.onnx_cpp2py_export.shape_inference.InferenceError: [ShapeInferenceError] (op_type:Transpose, node name: Transpose_7): [ShapeInferenceError] Inferred shape and existing shape differ in dimension 2: (64) vs (56)`

Looking at my original ONNX Resnet 18 model I do not understand whether this problem lies with my ONNX model or whether there is a problem with the qonnx import.. For the sake of clarity I have attached my model.

Any help would be very much appreciated.

Kind regards,

Meng Fei 
[resnet18_img_t.zip](https://github.com/Xilinx/finn/files/9548961/resnet18_img_t.zip)

**Comment by julienmas** [comment]
Hi Meng,

Could you send the onnx file you obtain just before you perform the transformation where you get the error?

**Reply by MFJI** [reply]
Hello julienmas,

Thank you so much for your reply. Here is the onnx file just before the transformation.
[step_tidy_up.zip](https://github.com/Xilinx/finn/files/9563690/step_tidy_up.zip)

**Reply by o-ibrahim33** [reply]
@julienmas any updates please?

**Reply by AliMirhosseini** [reply]
I have the same problem with this error 
InferenceError: [ShapeInferenceError] (op_type:Transpose, node name: Transpose_11): [ShapeInferenceError] Inferred shape and existing shape differ in dimension 2: (64) vs (32)
when running this line 
model = model.transform(Streamline())
did you get the answer? 

the model is resnet18 example in brevitas repo and the dataset is CIFAR-10.

**Reply by pankajan-T** [reply]
Hi, I've got the same issue with my custom resnet model as well. Is there a solution for this issue yet?
